scripts/bash_lib.sh

```diff
@@ -4,6 +4,8 @@
 # Include with:
 # . ./bash_lib.sh
 
+trap save_failure_logs ERR
```

Comment:
What will happen to the other failure points that don't need log collection? I suppose they will trigger log collection, reacting on the ERR signal of a command.

There are some code blocks which are not related to:
 - > installation / upgrade process of the server
 - > running queries to create or verify structures in the db

For example, blocks where we set-up the repository, make cache, query the repo and so on.

What I want to highlight is this most basic example:
```
set -e
trap save_failure_logs ERR

save_failure_logs() {
  trap "" ERR
  echo "I just saved the logs for Columnstore and MariaDB-Server"
}

echo "I am doing some pre-setup before Installing the Server"
echo "Presetup will fail, I have nothing installed. Do I really need to collect the logs?"

non_existent_command
```

Reply:
I think trap is a good option when it can provide a handle applicable to any failure point, allowing also for script expansion without worrying about its side effect.

Reply:
traps are set after initial setup when any of the following commands can be meaningfully triggered by a server fault. The set +e to disable is at the end of the block if there becomes a big of adminstration.

Even a missing command will be during the process and evaluating some aspect of the package is correct. If this is the case its an error also. Maybe the logs unneeded in this case, but the failure is still required.